samples/subsys/ipc/ipc_service/icmsg/boards/stm32h747i_disco.overlay

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2024 Celina Sophie Kalus <hello@celinakalus.de>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	chosen {
+		/delete-property/ zephyr,ipc_shm;
+	};
+
+	/* HSEM IPM is incompatible with HSEM MBOX */
+	/delete-node/ mailbox;
+
+	/* Define new memory regions for TX and RX */
+	/delete-node/ memory@38000000;
+
+	sram_tx: memory@38000000 {
+		reg = <0x38000000 0x08000>;
+	};
+
+	sram_rx: memory@38008000 {
+		reg = <0x38008000 0x08000>;
+	};
+};
+
+&pinctrl {
+	mbox: mbox@58026400 {
+		compatible = "st,mbox-stm32-hsem";
+		reg = < 0x58026400 0x400 >;
+		clocks = < &rcc 0xe0 0x2000000 >;
+		status = "okay";
+		interrupts = < 0x7d 0x0 >;
+		#mbox-cells = <1>;
```

Comment:
Since this is the same hw, I think we should use only one device tree description.
The SW implementation choice should be decided by Kconfig symbols.

In your case, it would mean that you would reuse the node and populate it with the new required properties which would be left unused by the legacy IPM driver.

Reply:
Note on this comment:
It does not prevent the Kconfig driver selection to depends on dts for instance with `DT_HAS_ZEPHYR_IPC_MSG_ENABLED`